crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -689,7 +694,9 @@ pub struct EntityLocation {
     pub archetype_id: ArchetypeId,
 
     /// The index of the entity in the archetype
-    pub index: usize,
+    pub archetype_index: u32,
```

Comment:
Can we have a comment on why this type is `repr(C)`? Adding new fields to types that we care about layouts of makes me nervous.

Reply:
Done.